web/src/test/java/com/microsoft/applicationinsights/web/internal/ThreadContextTests.java

```diff
@@ -78,6 +79,81 @@ public void run() {
         thread.join();
 
         Assert.assertNotNull(context[0]);
-        Assert.assertEquals(expectedRequestName, context[0].getHttpRequestTelemetry().getName());
+        Assert.assertSame(expectedRequestName, ThreadContext.getRequestTelemetryContext().getHttpRequestTelemetry()
+            .getName());
+    }
+
+    @Test
+    public void testChildThreadDoesnotGetContextUpdatedWhenrelyingonItls()
```

Comment:
It is hard to read the method name! Need to figure out a better syntax for readability.

Reply:
renamed